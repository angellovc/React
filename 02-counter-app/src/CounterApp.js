import React, {Fragment, useState} from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({value}) => {

  const [counter, setCounter] = useState(value);

  const handlerAdd = () => setCounter(counter + 1);
  const handlerSubs = () => setCounter(counter - 1)
  const handlerReset = () => setCounter(value);

  return (
    <Fragment>
      <h1>Counter App</h1>
      <h2>Number: {counter}</h2>
      <button onClick={handlerAdd} >+1</button>
      <button onClick={handlerReset}>Restart</button>
      <button onClick={handlerSubs}>-1</button>
    </Fragment>
  );
}
CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
}

export default CounterApp;


