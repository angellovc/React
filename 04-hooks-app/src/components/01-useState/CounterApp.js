import React, {useState} from 'react'
import './counter.css';

const CounterApp = () => {
  const [state, setcounter] = useState({
    counter1: 10,
    counter2: 20,
    counter3: 30,
    counter4: 40,
    counter5: 50,
  });
  const {counter1, counter2, counter3, counter4, counter5} = state;
  const handlerAdd1 = () => setcounter({
    ...state,
    counter1: counter1 + 1
  });
  return (
    <>
      <h2>Counter1: {counter1}</h2>
      <h2>Counter2: {counter2}</h2>
      <h2>Counter3: {counter3}</h2>
      <h2>Counter4: {counter4}</h2>
      <h2>Counter5: {counter5}</h2>
      <hr/>
      <button
        className="btn btn-primary"
        onClick={handlerAdd1}
      >+1 Counter1</button>
    </>
  )
}

export default CounterApp
