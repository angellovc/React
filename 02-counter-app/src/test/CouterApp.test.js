import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('<CounterApp /> Tests', () => {

    let wrapper = shallow(<CounterApp value={100} />);
    let value;
    beforeEach(() => {
        value = 100;
        wrapper = shallow(<CounterApp value={value} />);
    });

    test('Must be displayed correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('Default value must be equal to 100', () => {
        const textFromH2 = wrapper.find('h2').text();
        expect(textFromH2.split(' ')[1]).toBe(value.toString());
    });

    test('+1 button Must increase the counter', () => {
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text();
        expect(counterText).toBe('Number: 101');
    });

    test('-1 button must decrease the couter', () => {
        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text();
        expect(counterText).toBe('Number: 99');
    });

    test('reset button must restore default value which comes from props', () => {
        const wrapper = shallow(<CounterApp value={105} />);
        // add button
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        // reset button
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h2').text();
        expect(counterText).toBe('Number: 105');
    });

});