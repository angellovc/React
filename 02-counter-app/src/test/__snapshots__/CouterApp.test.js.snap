// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterApp /> Tests Must be displayed correctly 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    Number: 
    100
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Restart
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
